Compute value of float according to position value

gilles-leblanc opened the pull request:
According to CSS2 Section 9.7, if 'position' has a value of 'absolute'
or 'fixed' the computed value of 'float' should be 'none'.

This changes the float to a single_keyword_computed which checks the
positioned value of the element to compute the float value.

Fixes #8002

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/servo/servo/8111)

<!-- Reviewable:end -->

jdm commented:
There's a moratorium on creating new old-style reftests. Instead, you can use `./mach create-wpt --reftest tests/wpt/mozilla/tests/css/new_test.html --reference tests/wpt/mozilla/tests/css/new_test_reference.html` to automatically create a new-style one. However, I think this would be clearer as a regular web-platform-test instead (`./mach create-wpt tests/wpt/mozilla/tests/new_test.html`) which would just call `assert_equals` on the result of getComputedStyle.

gilles-leblanc committed:
Compute value of float according to position value

According to CSS2 Section 9.7, if 'position' has a value of 'absolute'
or 'fixed' the computed value of 'float' should be 'none'.

This changes the float to a single_keyword_computed which checks the
positioned value of the element to compute the float value.

Fixes #8002

gilles-leblanc commented:
I updated the pull request. 
Since the review hadn't begun proper I did a force push with the new version. I replaced the test witth a web-platform-test instead.

eefriedman commented on tests/wpt/mozilla/tests/css/float_relative_to_position.html:
```diff
@@ -0,0 +1,27 @@
+        var computed_style = getComputedStyle(document.querySelector("div"))["float"];
+        assert_equals(computed_style, "none");
+
+        document.querySelector("div").setAttribute("position", "absolute");
```
The `position` attribute on `<div>` has no meaning.  Did you mean to set the `style` attribute?

The fact that you could make this mistake and have the test still pass indicates that it's not ideal... maybe add a stylesheet with a style like `div { float: left }`, have the test check that float is left, then change the `position` using the style attribute.

gilles-leblanc committed:
Modify wpt test following review comment

gilles-leblanc commented:
Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

<sup>**[tests/wpt/mozilla/tests/css/float_relative_to_position.html, line 17 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8111#-K1CQrkFkUL3LEZUj3et-r1-17)** ([raw file](https://github.com/servo/servo/blob/bd471a7af2d9ddba7b3152947e2dfd839159961b/tests/wpt/mozilla/tests/css/float_relative_to_position.html#L17)):</sup>
Good point. I updated the pull request with a new commit which addresses this issue.
<!-- Reviewable comment -K1H2a9_4bieKba2VBjk:1932904617 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8111)
<!-- Sent from Reviewable.io -->

bors-servo commented:
:umbrella: The latest upstream changes (presumably #8152) made this pull request unmergeable. Please resolve the merge conflicts.

gilles-leblanc commented:
@jdm and @eefriedman Would you rather I rebase now (because of the changes mentioned by bors) and loose history in the review or would you rather I wait until the review is finished?

jdm commented:
Rebasing doesn't lose history. Reviewable is very permissive about these things.

jdm commented:
r? @SimonSapin

gilles-leblanc commented:
I know you guys are busy and this isn't urgent but I'm just pinging this to make it doesn't fall into the cracks :)

SimonSapin commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 595f11d has been approved by `SimonSapin`

<!-- @bors-servo r=SimonSapin 595f11dbd2a57dac9c650dd6eabc82f291bf6320 -->

SimonSapin commented:
(Sorry about the delay! I was away the last two weeks.)

bors-servo commented:
:hourglass: Testing commit 595f11d with merge 97389d7...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/193)

jdm commented:
@bors-servo: retry
* #8311

bors-servo commented:
:zap: Previous build results for [android](http://build.servo.org/builders/android/builds/177), [gonk](http://build.servo.org/builders/gonk/builds/182), [linux-dev](http://build.servo.org/builders/linux-dev/builds/176), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/189), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/199) are reusable. Rebuilding only [linux-rel](http://build.servo.org/builders/linux-rel/builds/193), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/187)...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/194)

gilles-leblanc commented:
Since it failed twice, I'll run it locally to check the result.

SimonSapin commented:
@bors-servo retry

* #7787

@gilles-leblanc, the failures are on different tests, both known to fail intermittently.

bors-servo commented:
:hourglass: Testing commit 595f11d with merge 5070c87...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/179), [gonk](http://build.servo.org/builders/gonk/builds/184), [linux-dev](http://build.servo.org/builders/linux-dev/builds/178), [linux-rel](http://build.servo.org/builders/linux-rel/builds/196), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/191), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/190), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/201)
